packages/legacy/src/index.js

```diff
@@ -0,0 +1,49 @@
+import { resolve } from 'path';
+
+const validName = /^[a-zA-Z_$][a-zA-Z$_0-9]*$/;
+
+export default function legacy(options) {
+  const exports = {};
+  Object.keys(options).forEach((file) => {
+    exports[resolve(file)] = options[file];
+  });
+
+  return {
+    name: 'legacy',
+
+    transform(content, id) {
+      if (id in exports) {
```

Comment:
return early for reducing nesting instead?

```suggestion
    if (!(id in exports) {
       return null;
    }
```

Reply:
I'm generally in favor of early returning as well, but we're trying to avoid optimizing and too many changes during the migration of a plugin. It's more about compatibility and laterally moving the plugin from one repo to this.